src/jit/gentree.h

```diff
@@ -1518,18 +1518,9 @@ struct GenTree
         return OperIsHWIntrinsic(gtOper);
     }
 
-#ifdef FEATURE_HW_INTRINSICS
-    inline bool OperIsSimdHWIntrinsic() const;
-#else
-    inline bool OperIsSimdHWIntrinsic() const
-    {
-        return false;
-    }
-#endif
-
-    bool OperIsSIMDorSimdHWintrinsic() const
+    bool OperIsSIMDorHWintrinsic() const
```

Comment:
Should this be cased as `OperIsSimdOrHWIntrinsic`?

Reply:
Probably. Will change.

Reply:
The JIT mainly uses "SIMD", not "Simd".

Reply:
> The JIT mainly uses "SIMD", not "Simd".

Yes, it looks that way. Grepping the codebase I get:
* 4843 hits for `SIMD`
* 1709 hits for `simd`
* 1701 hits for `Simd`

It would be nice if we could standardize on one. My preference would be for `Simd`, as it matches the casing specified in the general .NET Framework Design Guidelines and it makes it easier to read (although I don't feel particularly strongly on this point, given that it is native code; rather than managed).

Reply:
>  as it matches the casing specified in the general .NET Framework Design Guidelines

This isn't C# code.

> and it makes it easier to read

Well, if Simdor looks more readable than SIMDor…

> 1709 hits for simd

Those are variable names.

> 1701 hits for Simd

Those exist only because the people who worked on HW intrinsics ignored the already established naming.

Reply:
Personally I would prefer `SIMD`, but the JIT coding conventions say otherwise: https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#84-local-variables (that's for local variables, but my presumption is that applies generally to identifiers).

I believe that we have intentionally avoided undertaking any massive changes, but it makes sense to ensure that new or modified names are in conformance.

Reply:
> This isn't C# code.

I know, and I mentioned as such. My preference, however, is just towards a consistent casing scheme across both and to follow the existing well-defined rules that were already fairly extensively discussed/reviewed/etc.

> Well, if Simdor looks more readable than SIMDor…

The `Or` would also be capitilized (following standard pascal casing):
* `SimdOr`
* `SIMDOr`
* `Simdor`
* `SIMDor`

Reply:
> but the JIT coding conventions say otherwise

Ha ha, those aren't even consistent with .NET Framework Design Guidelines that says that if it has only 2 characters it's all upper case - `EEConfig` and not `EeConfig`.

Reply:
> but my presumption is that applies generally to identifiers

@CarolEidt, it looks to be explicitly covered under 8.9 (given this is a function): https://github.com/dotnet/coreclr/blob/master/Documentation/coding-guidelines/clr-jit-coding-conventions.md#89-functions-including-member-functions

Reply:
But that section isn't explicit about acronyms, though perhaps that's implied by "Pascal Casing" (I'm certainly not an expert in coding conventions).

Reply:
> Personally I would prefer SIMD,

Me too. This is not only about coding convention that is more about RyuJIT history... We have `V.N.Vectors` and `S.R.I.VectorXXX`,  and`V.N.Vectors` have used "SIMD" for a long time, so I named HW intrinsic related stuff in "Simd" at the very beginning. I know this is not a good style, but sometimes useful...

Reply:
I consider @BruceForstall to be the expert in the JIT coding conventions. I don't want to open up a discussion about this (I think we can all agree that such discussions quickly devolve), but it would be great if he could agree (or not) that `Simd` is what is specified in the conventions, and therefore wjat we should evolve to.

Reply:
The JIT coding conventions would say to use "Simd" in function and class names, but I view those as guidelines; I'm certainly willing to accept exceptions, if well motivated. (IMO, in this case I prefer @tannergooding 's `OperIsSimdOrHWIntrinsic` suggestion. `OperIsSimdOrHwIntrinsic` -- with a lower-case `w` for `Hw` looks odd.)